### PR TITLE
'liudmila_example_tests'

### PR DESCRIPTION
**Pull request by LiudmilaZhdanok**
has been done JAVA homework with tests

### CODE REVIEW

**Review by DenisAbramovDm — APPROVED** [review]
Lgfm

**Review by youRaFyodorov — DISMISSED** [review]
lgfm

**Review by Alexandr-R-94 — DISMISSED** [review]
lgfm

**Review by DenisAbramovDm — DISMISSED** [review]
Lgfm

**Review by Alexandr-R-94 — APPROVED** [review]
Lgfm

**Review by youRaFyodorov — APPROVED** [review]

**Comment by LiudmilaZhdanok** [comment]
I have marge problems. I'll come back to this later

File: homework/java/zhdanok_liudmila/lecture2/task1/src/main/java/by/jrr/mytests/bean/Dog.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package by.jrr.mytests.bean;
```
Comment (max-java) [inline homework/java/zhdanok_liudmila/lecture2/task1/src/main/java/by/jrr/mytests/bean/Dog.java:1]: It's a strange name for package.
Use tests only to mark smth related to tests.

File: homework/java/zhdanok_liudmila/lecture2/task1/src/main/java/by/jrr/mytests/bean/Dog.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+public class Dog {
+
+    String name;
+    int age;
+    double price;
```
Comment (max-java) [inline homework/java/zhdanok_liudmila/lecture2/task1/src/main/java/by/jrr/mytests/bean/Dog.java:7]: use private modificators

File: homework/java/zhdanok_liudmila/lecture2/task1/src/main/java/by/jrr/mytests/servise/DogServise.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
+package by.jrr.mytests.servise;
+
+import by.jrr.mytests.bean.Dog;
+
+public class DogServise {
+
+    Dog dog;
```
Comment (max-java) [inline homework/java/zhdanok_liudmila/lecture2/task1/src/main/java/by/jrr/mytests/servise/DogServise.java:7]: use private modificators

File: homework/java/zhdanok_liudmila/lecture2/task1/src/main/java/by/jrr/mytests/servise/DogServise.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+    public DogServise(Dog dog) {
+        this.dog = dog;
+    }
+
+    public String bark () {
+        return dog.getName() + " BARK!";
```
Comment (max-java) [inline homework/java/zhdanok_liudmila/lecture2/task1/src/main/java/by/jrr/mytests/servise/DogServise.java:14]: Usually we do not return calculation. For string it's better to prepare it in advance.
For String concatenations use method "".concat("BARK");

Best practices in return to use String.format("Dog %s says %s!", ownerName, dogVoice)

File: homework/java/zhdanok_liudmila/lecture2/task1/src/main/java/by/jrr/mytests/servise/DogServiseUtil.java [hunk 5]
```diff
@@ -0,0 +1,11 @@
+package by.jrr.mytests.servise;
+
+import by.jrr.mytests.bean.Dog;
+
+public class DogServiseUtil {
+
+
+    public static String bark (Dog dog) {
+        return dog.getName() + " BARK!";
```
Comment (max-java) [inline homework/java/zhdanok_liudmila/lecture2/task1/src/main/java/by/jrr/mytests/servise/DogServiseUtil.java:9]: Same as above

File: homework/java/zhdanok_liudmila/lecture2/task1/src/test/java/by/jrr/mytests/servise/DogServiseTest.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
+package by.jrr.mytests.servise;
```
Comment (max-java) [inline homework/java/zhdanok_liudmila/lecture2/task1/src/test/java/by/jrr/mytests/servise/DogServiseTest.java:1]: Don't use my by.jrr.
It's occupied!

File: homework/java/zhdanok_liudmila/lecture2/task1/src/main/java/by/jrr/mytests/servise/DogServise.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    }
+
+    public double convertPriceToHrivna(double exchangeRate) {
+        double convertedPrice = dog.getPrice()*exchangeRate;
```
Comment (integratorBY) [inline homework/java/zhdanok_liudmila/lecture2/task1/src/main/java/by/jrr/mytests/servise/DogServise.java:18]: use magic string. psfs for short key

File: homework/java/zhdanok_liudmila/lecture2/task1/src/test/java/by/jrr/mytests/servise/DogServiseTest.java [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+
+import by.jrr.mytests.bean.Dog;
+
+import org.junit.Test;
+
+import static org.junit.Assert.*;
```
Comment (max-java) [inline homework/java/zhdanok_liudmila/lecture2/task1/src/test/java/by/jrr/mytests/servise/DogServiseTest.java:7]: Don't use global static imports with *
 
 import static org.junit.Assert.assertEquals;

File: homework/java/zhdanok_liudmila/lecture2/task1/src/test/java/by/jrr/mytests/servise/DogServiseTest.java [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+
+public class DogServiseTest {
+
+    @Test
+    public void bark() {
+        Dog dog = new Dog();
```
Comment (max-java) [inline homework/java/zhdanok_liudmila/lecture2/task1/src/test/java/by/jrr/mytests/servise/DogServiseTest.java:13]: start from java 11 you can 
var dog = new Dog();

File: homework/java/zhdanok_liudmila/lecture2/task1/src/test/java/by/jrr/mytests/servise/DogServiseTest.java [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    @Test
+    public void convertPriceToHrivna_noDisconting() {
+        Dog dog = new Dog();
+        dog.setPrice(30);
+        double exchangeRate = 28.25;
+        double expected = 847.5;
+
+        DogServise dogServise = new DogServise(dog);
```
Comment (integratorBY) [inline homework/java/zhdanok_liudmila/lecture2/task1/src/test/java/by/jrr/mytests/servise/DogServiseTest.java:30]: use var

File: homework/java/zhdanok_liudmila/lecture2/task1/src/test/java/by/jrr/mytests/servise/DogServiseTest.java [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+    @Test
+    public void bark() {
+        Dog dog = new Dog();
+        dog.setName("Malysh");
+        String expected = "Malysh BARK!";
+
+        DogServise dogServise = new DogServise(dog);
+        String actual = dogServise.bark();
+
+        assertEquals(expected, actual );
```
Comment (max-java) [inline homework/java/zhdanok_liudmila/lecture2/task1/src/test/java/by/jrr/mytests/servise/DogServiseTest.java:20]: remove space in the end

File: homework/java/zhdanok_liudmila/lecture2/task1/src/test/java/by/jrr/mytests/servise/DogServiseTest.java (hunk 9 above)
Comment (max-java) [inline homework/java/zhdanok_liudmila/lecture2/task1/src/test/java/by/jrr/mytests/servise/DogServiseTest.java:13]: you can use private method to construct Dog

```
private Dog makeDogWithPrice(Double price) {
      Dog dog = new Dog();
      dog.setName("Malysh");
      dog.setPrice(price);
      return dog;
}
```

here you can use makeDogWithPrice(null);

**Review by integratorBY — APPROVED** [review]
`looks good for me
`

File: homework/java/zhdanok_liudmila/lecture2/task1/src/test/java/by/jrr/mytests/servise/DogServiseTest.java [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+
+        assertEquals(expected, actual, 0.01);
+    }
+
+    @Test
+    public void convertPriceToHrivna_chouldReturnWithDiscount() {
+        Dog dog = new Dog();
+        dog.setPrice(97);
+        double exchangeRate = 28.25;
+        double expected = 2658.04; //2658.0425
```
Comment (max-java) [inline homework/java/zhdanok_liudmila/lecture2/task1/src/test/java/by/jrr/mytests/servise/DogServiseTest.java:41]: remove unnecessary comment

File: homework/java/zhdanok_liudmila/lecture2/task1/src/test/java/by/jrr/mytests/servise/DogServiseTest.java [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+        double expected = 847.5;
+
+        DogServise dogServise = new DogServise(dog);
+        double actual = dogServise.convertPriceToHrivna(exchangeRate);
+
+        assertEquals(expected, actual, 0.01);
+    }
+
+    @Test
+    public void convertPriceToHrivna_chouldReturnWithDiscount() {
```
Comment (max-java) [inline homework/java/zhdanok_liudmila/lecture2/task1/src/test/java/by/jrr/mytests/servise/DogServiseTest.java:37]: _chould

mistype?

File: homework/java/zhdanok_liudmila/lecture2/task1/src/test/java/by/jrr/mytests/servise/DogServiseUtilTest.java [hunk 14]
```diff
@@ -0,0 +1,20 @@
+package by.jrr.mytests.servise;
+
+import by.jrr.mytests.bean.Dog;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
```
Comment (max-java) [inline homework/java/zhdanok_liudmila/lecture2/task1/src/test/java/by/jrr/mytests/servise/DogServiseUtilTest.java:6]: remove global import *

File: homework/java/zhdanok_liudmila/lecture2/task1/src/main/java/by/jrr/mytests/Application.java [hunk 15]
```diff
@@ -0,0 +1,12 @@
+package by.jrr.mytests;
+
+
+
+public class Application {
+
+    public static void main(String[] args) throws Exception {
```
Comment (max-java) [inline homework/java/zhdanok_liudmila/lecture2/task1/src/main/java/by/jrr/mytests/Application.java:7]: Here you should place at least comment why main method is blank
Exception not needed

**Review by kashlyak — COMMENTED** [review]
lgfm

**Review by youRaFyodorov — APPROVED** [review]
lgfm

**Review by Dimorolll — APPROVED** [review]
lgfm

**Pull request merged**